src/DynamicData/Binding/SortAndBind.cs

```diff
@@ -0,0 +1,160 @@
+﻿// Copyright (c) 2011-2023 Roland Pheasant. All rights reserved.
+// Roland Pheasant licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for full license information.
+
+using System.Reactive.Linq;
+using DynamicData.Cache;
+using DynamicData.Cache.Internal;
+
+namespace DynamicData.Binding;
+
+/*
+ * A much more optimised bind where the sort forms part of the binding.
+ *
+ * This is much efficient as the prior sort mechanism would resort and clone the entire
+ * collection upon every change in order that the sorted list could be transmitted to the bind operator.
+ *
+ */
+internal sealed class SortAndBind<TObject, TKey>(
+    IObservable<IChangeSet<TObject, TKey>> source,
+    IComparer<TObject> comparer,
+    SortAndBindOptions options,
+    IList<TObject> target)
+    where TObject : notnull
+    where TKey : notnull
+{
+    private readonly Cache<TObject, TKey> _cache = new();
+    private readonly object _locker = new();
+
+    public IObservable<IChangeSet<TObject, TKey>> Run() =>
+        source
+            .Synchronize(_locker)
+            // apply sorting as a side effect of the observable stream.
+            .Do(changes =>
+            {
+                // clone to local cache so that we can sort entire set when threshold is over a certain size.
+                _cache.Clone(changes);
+
+                // apply sorted changes to the target collection
+                if (target.Count == 0 || (options.ResetThreshold != 0 && options.ResetThreshold < changes.Count))
```

Comment:
Its been that way since day one, and very few people have ever even noticed that behaviour exists so I suggest why change a proven formula.

However, that does not mean that we can't look into these extra options as a later improvement. Personally I do not want that do extra at this stage because

1. I need to apply the same treatment to visualise and to page as they are the only other operators that rely on sort
2. The improvements are already missive and solved a serious problem which DD currently has.

I suggest one this is merged, you can raise an issue to make these suggestions